### PR TITLE
additions for scrubbers, grammar fixes

### PR DESCRIPTION
**Pull request by seankane-msft**
Adding a section about scrubbing/purging secrets from test recording files. Made a few grammar fixes/tweaks to packaging and tests-advanced files as well.

### CODE REVIEW

File: doc/dev/tests.md [hunk 1]
```diff
@@ -266,6 +267,14 @@ From your terminal run the `pytest` command to run all the tests that you have w
...
 
+### Purging Secrets
+
+The `yaml` files created from running tests in live mode store the request and response interactions between the library and the service and this can include authorization, account names, shared access signatures, and other secrets. The recordings are included in our public GitHub repository, making it important for us to remove any secrets from these recordings before committing them to the repository. There are two easy ways to remove secrets. The first is the `PowerShellPreparer` implementation, discussed above. This method will automatically purge the keys with the provided fake values. The second way is to use the `self.scrubber.register_name_pair(key, fake_key)` method, which is used when a secret is dynamically created during a test. For example, [KeyVault](https://github.com/Azure/azure-sdk-for-python/blob/9091e0ea0ddadc349cdd0f92579e155c7f44220b/sdk/keyvault/azure-keyvault-administration/tests/test_access_control.py#L43-L56) uses this method to remove GUIDs and service principals from recordings.
```
Comment (chlowell) [inline doc/dev/tests.md:272]: This Key Vault code shows one way to replace values in recordings but these particular values aren't secret, just non-deterministic. Similar solution, different problem. One of the processors in azure-devtools, like [SubscriptionRecordingProcessor](https://github.com/Azure/azure-sdk-for-python/blob/ad65f699fafcf80949e84b21b74fe8ea67ddd0dc/tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py#L33), might make a better example here.

Comment (rakshith91) [reply doc/dev/tests.md:272]: One other way for scrubbing:
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py#L31

Comment (kristapratico) [inline doc/dev/tests.md:272]: Might be worth noting that `self.scrubber.register_name_pair(key, fake_key)` comes from AzureTestCase

File: doc/dev/packaging.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-Notes:
+Note:
 - This article is not about setup.py or setup.cfg or the right way to *write* the packaging, it's about what instructions you should use to achieve this. If you are fluent in setuptools, and prefer to write the suggestions in setup.cfg and not in setup.py, this is not a concern.
 
 # What are the constraints?
 
 We want to build sdist and wheels in order to follow the following constraints:
 - Solution should work with *recent* versions of pip and setuptools (not the very latest only, but not archaeology either)
-- Wheels must work with Python 2.7 and 3.4+
+- Wheels must work with Python 2.7 and 3.5+
```
Comment (swathipil) [inline doc/dev/packaging.md:22]: Will this be updated to 3.6+ from next release on, or does that not affect this?

File: doc/dev/tests-advanced.md [hunk 3]
```diff
@@ -169,11 +169,11 @@ Notes:
 
 ## mgmt_settings_real file
 
-Instead of using a `.env` file, you can use the `mgmt_settings_real.py` file by copying `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py` to `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_real.py` and providing real credentials to it. To
+A `mgmt_settings_real.py` can be used in place of a `.env` file by copying `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py` to `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_real.py` and providing real credentials to it.
```
Comment (swathipil) [inline doc/dev/tests-advanced.md:172]: For clarity, something like:
```suggestion
A `mgmt_settings_real.py` can be used in place of a `.env` file by copying `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py` to `sdk/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_real.py` and providing real credentials to it. The following changes should be made to the `mgmt_settings_real.py` file.
```

File: doc/dev/tests-advanced.md [hunk 4]
```diff
@@ -169,11 +169,11 @@ Notes:
...
+1. Change the value of the `SUBSCRIPTION_ID` variable to your organizations subscription ID, which can be found in the "Overview" section of the "Subscriptions" blade in the [Azure portal](https://portal.azure.com/).
+2. Defining `TENANT_ID`, `CLIENT_ID`, and `CLIENT_SECRET` which are available after creating a Service Principal or can be retrieved from the Azure Portal after creating a Service Principal. Check out the [Azure docs](https://docs.microsoft.com/cli/azure/ad/sp?view=azure-cli-latest#az_ad_sp_create_for_rbac) to create  a Service Principal with a simple one line command to create one. The recommended practice is to include your alias or name in the Service Principal name.
+3. Change the `get_azure_core_credentials(**kwargs):` function to construct and return a `ClientSecretCredential` object. The `client_id`, `client_secret`, `tenant_id` are provided when you create a service principal. This method should look like this:
```
Comment (swathipil) [inline doc/dev/tests-advanced.md:176]: seems to be a little bit of redundancy, maybe change to something like this?:
```suggestion
2. Define `TENANT_ID`, `CLIENT_ID`, and `CLIENT_SECRET`, which can be retrieved from the Azure Portal after creating a Service Principal. Check out the [Azure docs](https://docs.microsoft.com/cli/azure/ad/sp?view=azure-cli-latest#az_ad_sp_create_for_rbac) to create a Service Principal with a simple one line command. The recommended practice is to include your alias or name in the Service Principal name.
3. Change the [`get_azure_core_credentials(**kwargs)`](https://github.com/Azure/azure-sdk-for-python/blob/7dd62a6900ea052d244c38c9d48de7c53f67defa/tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py#L39) function in the `mgmt_settings_real.py` file to construct and return a `ClientSecretCredential` object. Pass in the `CLIENT_ID`, `CLIENT_SECRET`, `TENANT_ID` values to the `ClientSecretCredential` object. This method should look like this:
```

**Review by swathipil — COMMENTED** [review]
small nits, changes may or may not be applied. lmk if any of the comments look wrong :)

File: doc/dev/packaging.md (hunk 2 above)
Comment (seankane-msft) [reply doc/dev/packaging.md:22]: Updating to 3.6+

File: doc/dev/tests.md (hunk 1 above)
Comment (seankane-msft) [reply doc/dev/tests.md:272]: Combined the two of these by showing off the cosmos preparer.

**Review by kristapratico — APPROVED** [review]

**Pull request merged**